src/stream.js

```diff
@@ -21,8 +26,63 @@ module.exports.handler = async (event) => {
 	delete(groupedRecords['deleteMe']); // delete bad events
 
 	for (const records of Object.entries(groupedRecords)) {
-		const latestRecord = records.pop();
+		logger.debug({records}, 'Records');
+		const latestRecord = records.pop().pop();
+		// Tag for a transient record that should be ignored.
+		if (latestRecord.date === '_current') {
+			continue;
+		}
 		logger.debug({latestRecord}, 'Event received');
-	}
 
+		const responses = latestRecord.users && Object.keys(latestRecord.users).map(key => {
+			return {
+				name: `<@${key}>`,
+				answer: latestRecord.users[key] && latestRecord.users[key].answer  === 'fact'
+			};
+		});
+
+		const blockKitParamters = {
+			date: Date.parse(latestRecord.date),
+			questionText: latestRecord.questionText,
+			responses,
+			answer: latestRecord.answer,
+			answerText: latestRecord.answerText
+		};
+
+		const blocks = latestRecord.answer
+			? blockKit.getClosedMessage(blockKitParamters)
+			: blockKit.getOpenMessage(blockKitParamters);
+
+		const postMessageParamters = {
+			date: latestRecord.date,
+			channel: latestRecord.channel,
+			text: '',
+			blocks,
+			ts: latestRecord.slackId
+		};
+
+		const result = latestRecord.slackId
+			? await slack.chat.update(postMessageParamters)
+			: await postMessage(postMessageParamters);
```

Comment:
You could move this logic into `postMessage` to make it behave like an upsert so that it internally figures out whether to call `slack.chat.postMessage` or `slack.chat.update`. The minor benefit would be this calling code gets slightly simpler.